engine/class_modules/sc_demon_hunter.cpp

```diff
@@ -2371,6 +2371,25 @@ struct fel_devastation_t : public demon_hunter_spell_t
     }
   }
 
+  void last_tick( dot_t* d ) override
+  {
+    demon_hunter_spell_t::last_tick( d );
+
+    if ( p()->talent.vengeance.darkglare_boon->ok() )
+    {
+      // CDR reduction and Fury refund are separate rolls per Realz
+      timespan_t minimum_cdr_reduction = timespan_t::from_seconds(p()->talent.vengeance.darkglare_boon->effectN(1).base_value());
+      timespan_t maximum_cdr_reduction = timespan_t::from_seconds(p()->talent.vengeance.darkglare_boon->effectN(2).base_value());
+      timespan_t cdr_reduction = rng().range(minimum_cdr_reduction, maximum_cdr_reduction);
+      double minimum_fury_refund = p()->talent.vengeance.darkglare_boon->effectN(3).base_value();
+      double maximum_fury_refund = p()->talent.vengeance.darkglare_boon->effectN(4).base_value();
+      double fury_refund = rng().range(minimum_fury_refund, maximum_fury_refund);
+
+      p()->cooldown.fel_devastation->adjust(cdr_reduction);
```

Comment:
Isn't the adjustment value coming from spell data the percentage of the total cooldown reduced?

Reply:
That is correct, not a flat value; I'll fix it.

Reply:
Just a note that a flat value ends up working out atm due to Fel Dev CD being 1min by default.